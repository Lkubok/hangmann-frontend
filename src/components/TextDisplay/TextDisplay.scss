@import "../../scss/mixins.scss";
@import "../../scss/variables.scss";


.test-app-hoc{
    display: flex;
    flex-flow: row;
    justify-content: center;
    align-items: center;
    height: 30px;
    background-color:$dark-color-light-hover;
    color: $white-color;
    color: $primary-color;
    white-space: pre;
    span{
        color: $secondary-color;
    }
}
