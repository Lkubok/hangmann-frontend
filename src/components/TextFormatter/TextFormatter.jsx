import React, { Component } from "react";

export default function TextFormatter(WrappedComponent) {
  return class TextFormatter extends Component {
    render() {
      const { text } = this.props;
      return <WrappedComponent text={text} version={" v_1.0"} />;
    }
  };
}
