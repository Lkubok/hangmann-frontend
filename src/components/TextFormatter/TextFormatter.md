
## TextFormatter

it expects as argument another component that will display data from props.data

### Example of use 

```

export class TextDisplay extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return <div className="test-app-hoc">Formatted text: {this.props.text}</div>;
  }
}

export default TextFormatter(TextDisplay);
```




