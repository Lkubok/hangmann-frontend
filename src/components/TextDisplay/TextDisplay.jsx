import React, { Component } from "react";
import TextFormatter from "../TextFormatter";

import "./TextDisplay.scss";

export class TextDisplay extends Component {
  render() {
    const { text, version } = this.props;
    return (
      <div className="test-app-hoc">
        {text}
        <span>{version}</span>
      </div>
    );
  }
}

export default TextFormatter(TextDisplay);
