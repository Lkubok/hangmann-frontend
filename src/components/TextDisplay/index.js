import TextDisplay from "./TextDisplay";
export default TextDisplay;
