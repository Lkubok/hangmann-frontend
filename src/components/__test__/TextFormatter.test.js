import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { TextDisplay } from "../TextDisplay/TextDisplay";
import TextFormatter from "../TextFormatter";
Enzyme.configure({ adapter: new Adapter() });

describe("HoC TextFormatter component", () => {
  it("It should add version prop to TextDisplay component", () => {
    const wrapper = mount(<TextDisplay text="test" />);
    expect(wrapper.props().text).toEqual("test");
  });

  it("Should match snapshot", () => {
    const wrapper = mount(<TextDisplay text="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("It should add version prop to TextDisplay component", () => {
    const TestedComponent = TextFormatter(TextDisplay);
    const wrapper = shallow(<TestedComponent />);
    expect(wrapper.props().version).toEqual(" v_1.0");
  });
  it("should receive text prop", () => {
    const Element = () => <div text="test" />;
    const TestedComponent = TextFormatter(Element);
    const wrapper = shallow(<TestedComponent text={"test"} />);
    wrapper.instance().forceUpdate();
    expect(wrapper.props().text).toEqual("test");
  });
});
