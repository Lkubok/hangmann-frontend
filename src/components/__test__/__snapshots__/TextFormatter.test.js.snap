// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoC TextFormatter component Should match snapshot 1`] = `
<TextDisplay
  text="test"
>
  <div
    className="test-app-hoc"
  >
    test
    <span />
  </div>
</TextDisplay>
`;
